### PR TITLE
fix(eslint-plugin): [prefer-includes] ignore option chaining before indexOfs

### LINKED ISSUE #3341: [prefer-includes] Auto-fix incorrect when optional chaining
- [ x ] I have tried restarting my IDE and the issue persists.
- [ x ] I have updated to the latest version of the packages.
- [ x ] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/prefer-includes": "error"
  }
}
```

```TS
let x: string;
if (x?.indexOf("y") === -1) {
	console.log("y not found in x");
}
```

**Expected Result**
Instance is either not auto-fixed or fixed correctly honoring the optional chaining

**Actual Result**
Auto-fix changes logic.  Result for example below.  The if statement returns false originally and true after the auto-fix.
```TS
let x: string;
if (!x?.includes("y")) {
	console.log("y not found in x");
}
```

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.22.0` |
| `@typescript-eslint/parser`        | `4.22.0` |
| `TypeScript`                       | `4.2.4` |
| `ESLint`                           | `7.25.0` |
| `node`                             | `12.19.0` |

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
Fixes #3341

I was on the fence of whether to just remove the fixer for `?.indexOf` instead, but went with removing it altogether because of the subtle behavior differences:

<table>
<thead>
<th>Original</th>
<td><code>x = undefined</code></td>
<td><code>x = ""</code></td>
<td><code>x = "y"</code></td>
</thead>
<tbody>
<tr>
<th><code>x?.indexOf("y") === -1</code></th>
<td><code>false</code></td>
<td><code>true</code></td>
<td><code>false</code></td>
</tr>
<tr>
<td><code>!x?.includes("y")</code></td>
<td><code>true</code> ❌</td>
<td><code>true</code></td>
<td><code>false</code></td>
</tr>
<tr>
<th><code>x?.indexOf("y") !== -1</code></th>
<td><code>true</code></td>
<td><code>false</code></td>
<td><code>true</code></td>
</tr>
<tr>
<td><code>x?.includes("y")</code></td>
<td><code>undefined</code> ❌</td>
<td><code>false</code></td>
<td><code>true</code></td>
</tr>
</tbody>
</table>

### CODE REVIEW

**Review by bradzacher — COMMENTED** [review]
Question: Do we want to keep the reporting but remove the fixer only?

I.e. the user has to evaluate and fix the code as required

File: packages/eslint-plugin/tests/rules/prefer-includes.test.ts [hunk 1]
```diff
@@ -13,165 +11,140 @@ const ruleTester = new RuleTester({
...
-      acc.push({
-        ...c,
-        code: c.code.replace('.', '?.'),
-        output: 'output' in c ? c.output?.replace('.', '?.') : null,
-      });
-    }
-
-    return acc;
-  }, []);
-}
```
Comment (JoshuaKGoldberg) [inline packages/eslint-plugin/tests/rules/prefer-includes.test.ts:39]: This `addOptional` wrapper was hiding the test cases from being checked by Prettier... although I could keep it in and remove the `output` field, it does feel weird to me to auto-generate them this way when the only difference between the two is checking `callNode.parent.parent` vs. `callNode.parent`?

Comment (bradzacher) [reply packages/eslint-plugin/tests/rules/prefer-includes.test.ts:39]: this was just my lazy way of covering all the cases 😅

**Review by bradzacher — APPROVED** [review]
LGTM - thanks for fixing this!

**Pull request merged**